### ISSUE TITLE
Data augmentation

### ISSUE THREAD

**Issue by santi-rodriguez** [LEB-EPFL/MitoSplit-Net#3]
Add functions to **scale, rotate and mirror** the input data. 

I was talking with Chen, and he thinks that training the network using only the above-mentioned transformations could bias the model to detect only midzone cuts. To increase the robustness of the model, I propose to add a **nonlinear stretching**, so the cuts locate at different points along the longest axis of the mitochondria.

What is your opinion?

**Comment by wl-stepp** [comment]
I don't think that those transformations can change where the fissions happen in the training data. I think a peripheral fission will still look like a peripheral fission to the network. The transformations are full frame, while the feature for a fission are much more localized. So I don't worry that there might be a bias due to the data augmentation to central fissions.

But there are things to consider...
**Scale**: Do we clean up the ground truth again after scaling, to make sure that nothing is directly on the edge of the frame?
**Rotate**: How is this usually done? By padding and recropping or would you only rotate scaled images?

**Comment by santi-rodriguez** [comment]
I added a function to training.py to [train the network with augmented data on the fly](https://github.com/LEB-EPFL/MitoSplit-Net/blob/2438d2b64038ef54c58e110333051c6de1c68495/mitosplit-net/training.py#L142) based on the [Augmentor library](https://github.com/mdbloice/Augmentor).  I like it because it is easy to use, as the main object is a Pipeline which you can extend using as many operations as you want. 

There are a lot of transformations available, and it is even possible to define your own operations. For example, I included an operation that clears the border of the image before being fed to the network.

This is still being tested, so I will keep posting new updates during this week.

**Comment by wl-stepp** [comment]
Nice, the doubts about how exactly to implement some of the transformations here seem to be already implemented and answered in the examples on the augmentor module homepage.

**Comment by santi-rodriguez** [comment]
Indeed, augmentor has very detailed documentation. Unfortunately, it converts inputs and outputs to PIL images instead of working with numpy arrays. Unless you correct that manually, this leads to incompatibilities with TensorFlow that produce errors during training. Finally, I moved to [albumentations](https://github.com/albumentations-team/albumentations), which let me do the same using only arrays. Updates are coming soon.
